backend/src/main/java/de/bund/digitalservice/ris/search/utils/LdmlTemporalData.java

```diff
@@ -0,0 +1,97 @@
+package de.bund.digitalservice.ris.search.utils;
+
+import de.bund.digitalservice.ris.search.caselawhandover.shared.XmlUtils;
+import de.bund.digitalservice.ris.search.models.ldml.TimeInterval;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+public class LdmlTemporalData {
+
+  private static final Logger logger = LogManager.getLogger(LdmlTemporalData.class);
+
+  private static final String X_PATH_ALL_EVENT_REF = "//*[local-name()='eventRef']";
```

Comment:
While we're at it, could we refine this XPath to point to the actual node in the document instead of traversing the whole tree?